src/igr-tests/environ2/run-test.sh

```diff
@@ -14,11 +14,11 @@ RGID=$(id -g)
 for var in USER LOGNAME SHELL; do
     unset $var
 done
-# $UID & $GID are readonly on MacOS
-if [ "$(uname)" != "Darwin" ]; then
-    unset UID
-    unset GID
-fi
+# ignore $UID & $GID unsetting errors
+# Some shells make them readonly
```

Comment:
It would've been good to be specific here. Eg. "some shells - eg bash - make them readonly".

Reply:
It actually based on system configuration for example I easily unset it on bash:
```bash
> echo $GID
1000
> unset GID
> echo $GID

>
```

Reply:
(or maybe bash have different behavior in non-interactive mode?)

Reply:
UID is readonly in bash. Bash doesn't set GID at all.
On MacOS, zsh sets UID and GID readonly.